### PR TITLE
Vpn support

### PR DESCRIPTION
**Pull request by mjz-smu**
- Adds VPN service to manage IPSec gateways and tunnels on cloud providers
- Includes GCP and AWS provider implementations in this PR. 
- Adds `cidr` flag to vm_group to allow for custom subnets.
Related ticket:  [#880](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/issues/880)
[Usage](https://gist.github.com/mjz-smu/ac740cfb5bf8ad1ae22eb24d140c7f6b)

### CODE REVIEW

**Comment by yuyantingzero** [comment]
@mjz-smu thanks for contribution, is it possible to break into smaller commits to speed up review process and increase readability?

File: perfkitbenchmarker/linux_benchmarks/iperf_vpn_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
+# Copyright 2014 PerfKitBenchmarker Authors. All rights reserved.
```
Comment (s-deitz) [inline perfkitbenchmarker/linux_benchmarks/iperf_vpn_benchmark.py]: Related to the request to break this into smaller PRs. This benchmark could be separated out as a follow-on.

File: perfkitbenchmarker/linux_virtual_machine.py [hunk 2]
```diff
@@ -494,6 +494,11 @@ def RecordAdditionalMetadata(self):
     self.os_metadata['kernel_release'] = self.kernel_release
     if FLAGS.append_kernel_command_line:
       self.os_metadata['kernel_command_line'] = self.kernel_command_line
+    self.os_metadata['cpu_family'] = lscpu_results.cpu_family
```
Comment (s-deitz) [inline perfkitbenchmarker/linux_virtual_machine.py]: Similarly, this additional metadata could be a separate change.

File: perfkitbenchmarker/linux_benchmarks/iperf_vpn_benchmark.py (hunk 1 above)
Comment (mjz-smu) [reply perfkitbenchmarker/linux_benchmarks/iperf_vpn_benchmark.py]: Sorry for the bloated PR. This started as a single benchmark and grew a bit big. 
To clarify - is the preference to split into smaller commits, smaller PRs, or both? 
Logical dependencies are: 
|custom CIDR| <-- |VPN svc| <-- |VPN providers| <-- |VPN benchmark|
If these work better as separate PRs I'll close this one and resubmit separately in the above order.

Comment (yuyantingzero) [reply perfkitbenchmarker/linux_benchmarks/iperf_vpn_benchmark.py]: Sorry I mean smaller PRs. 
This order makes a lot of sense

Comment (mjz-smu) [reply perfkitbenchmarker/linux_benchmarks/iperf_vpn_benchmark.py]: Thanks! Closing this in favour of the smaller PRs then...

**Pull request closed without merging**